### ISSUE TITLE
Updating Disk Cache on Every Import

### ISSUE THREAD

**Issue by firefox15** [ecitsolutions/Autotask#74]
Unfortunately, I am still having the issue as specified in #66 .  I cannot get the module to respect the cache.  It populates the cache, but then it just deletes it the next time it runs.  This used to work fine, so I have no idea why this is happening:

I have tried both version 1.6.2.17 and 1.6.4.2 without success.  I have manually deleted the cache as well with no change.  Every time I run it, it simply wants to recreate it.  Even after I run it, it rebuilds it _again_ after I run `Connect-AtwsWebAPI` to connect.  Any ideas?

```
DEBUG: Autotask.psm1: Personal disk cache: Found 196 script files in C:\Users\Alex\Documents\WindowsPowershell\Cache\300371\Dynamic                                      

DEBUG: Autotask.psm1: Personal disk cache: Wrong number of script files or scripts are not the right version in C:\Users\Alex\Documents\WindowsPowershell\Cache\300371\Dynamic, refreshing all entities. 
```

**Comment by klemmestad** [comment]
Thank you for posting this issue! You gave me the missing piece of the puzzle. I am convinced this happens because the API version of your Autotask tenant has been upgraded to a newer version than the module was built against. Our tenant is scheduled to be upgraded to night, so tomorrow I will have the same problem as you and be able to debug and hopefully solve this for real this time.

**Comment by firefox15** [comment]
Thanks so much @klemmestad!  Hopefully this helps you solve it!

**Comment by klemmestad** [comment]
While you wait for my fix you may want to repair your downloaded version like this:

Change <install folder>\Autotask\1.6.4.2\Autotask.psm1:line 179
```powershell
if ($ScriptVersion.Count -ne $FunctionCount) {
```
To this
```powershell
if ($ScriptVersion.Count -ge $FunctionCount) {
```
Because a new API version may have more dynamic entities than the previous one the count should be allowed to be larger than or equal to.

**Comment by firefox15** [comment]
That definitely did the trick.  Thanks so much for your hard work on this and getting back to me so quickly.  It's a very helpful module.

**Comment by klemmestad** [comment]
Should be fixed in version 1.6.5 (released)
